apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts

```diff
@@ -0,0 +1,67 @@
+import { db, eq, schema } from "@/lib/db";
+import { env } from "@/lib/env";
+import { ingestAuditLogs } from "@/lib/tinybird";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { auth, t } from "../../trpc";
+
+export const updateRootKeyName = t.procedure
+  .use(auth)
+  .input(
+    z.object({
+      keyId: z.string(),
+      name: z.string().nullish(),
+    }),
+  )
+  .mutation(async ({ input, ctx }) => {
+    const key = await db.query.keys.findFirst({
+      where: (table, { eq, isNull, and }) =>
+        and(eq(table.id, input.keyId), isNull(table.deletedAt)),
+      with: {
+        workspace: true,
+      },
+    });
+
+    if (!key || key.ownerId !== ctx.tenant.id) {
```

Comment:
Let's do look up the workspace and make sure the rootKey belongs to the `for_workspace_id`  so we can be 100% sure that the key belongs. 
```
const workspace = await db.query.workspaces.findFirst({
      where: (table, { and, eq, isNull }) =>
        and(eq(table.tenantId, ctx.tenant.id), isNull(table.deletedAt)),
    });
    if (!workspace) {
      throw new TRPCError({
        code: "NOT_FOUND",
        message:
          "We are unable to find the correct workspace. Please contact support using support@unkey.dev.",
      });
    }
````

Then do 

if(!key || key.forWorkspaceId !== workspace.id